tests/Doctrine/Tests/ORM/Hydration/ObjectHydratorTest.php

```diff
@@ -1021,18 +1024,21 @@ public function testCreatesProxyForLazyLoadingWithForeignKeys(): void
             ],
         ];
 
-        $proxyInstance = new ECommerceShipping();
-
-        // mocking the proxy factory
-        $proxyFactory = $this->getMockBuilder(ProxyFactory::class)
-                             ->onlyMethods(['getProxy'])
-                             ->disableOriginalConstructor()
-                             ->getMock();
-
-        $proxyFactory->expects(self::once())
-                     ->method('getProxy')
-                     ->with(self::equalTo(ECommerceShipping::class), ['id' => 42])
-                     ->will(self::returnValue($proxyInstance));
+        // extending the proxy factory to spy on getProxy()
+        $proxyFactory = new class (
+            $this->entityManager,
+            sys_get_temp_dir(),
+            'Proxies',
+            ProxyFactory::AUTOGENERATE_ALWAYS,
+        ) extends ProxyFactory {
+            public function getProxy(string $className, array $identifier): InternalProxy
+            {
+                TestCase::assertSame(ECommerceShipping::class, $className);
+                TestCase::assertSame(['id' => 42], $identifier);
+
+                return parent::getProxy($className, $identifier);
+            }
+        };
```

Comment:
@derrabus I pushed a new commit, it reworks this test and the one below, allowing to use a native return type declaration for `ProxyFactory::getProxy()`